frontend/packages/console-shared/src/components/formik-fields/EnvironmentField.tsx

```diff
@@ -1,27 +1,75 @@
 import * as React from 'react';
+import * as _ from 'lodash';
 import { useFormikContext, FormikValues } from 'formik';
-import { EnvironmentPage } from '@console/internal/components/environment';
+import { NameValueEditor } from '@console/internal/components/utils/name-value-editor';
 import { FormGroup } from '@patternfly/react-core';
-import { EnvironmentFieldProps, NameValuePair, NameValueFromPair } from './field-types';
+import { EnvironmentFieldProps } from './field-types';
 import { getFieldId } from './field-utils';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { k8sGet } from '@console/internal/module/k8s';
 
 const EnvironmentField: React.FC<EnvironmentFieldProps> = ({
   label,
   helpText,
   required,
+  envs,
   ...props
 }) => {
   const { setFieldValue } = useFormikContext<FormikValues>();
   const fieldId = getFieldId(props.name, 'env-input');
+  const [nameValue, setNameValue] = React.useState(!_.isEmpty(envs) ? envs : [['', '']]);
+  const [configMaps, setConfigMaps] = React.useState({});
+  const [secrets, setSecrets] = React.useState({});
+  const nsRef = React.useRef({ namespace: props.obj?.metadata?.namespace });
+  nsRef.current = { namespace: props.obj?.metadata?.namespace };
```

Comment:
This is unnecessary. You could have just put the namespace as a dependency.

Reply:
Done.